pkg/httpd/simpleserver.go

```diff
@@ -48,12 +49,26 @@ func (simpleServer *SimpleServer) Start() error {
 	mux.HandleFunc("/healthz", webhook.HealthCheckHandler)
 	mux.HandleFunc("/mutate", admissionHandler.HandleAdmission)
 
+	addr := ":9090"
+	startMetricsServer(addr)
+
 	if simpleServer.Local {
 		return server.ListenAndServe()
 	}
 	return server.ListenAndServeTLS(simpleServer.CertFile, simpleServer.KeyFile)
 }
 
+func startMetricsServer(addr string) {
+	log.Infoln(fmt.Sprintf("Starting metrics server on %s", addr))
+	metricsRouter := http.NewServeMux()
+	metricsRouter.Handle("/metrics", promhttp.Handler())
+	metricsServer := &http.Server{
+		Addr:    addr,
+		Handler: metricsRouter,
+	}
+	log.Fatal(metricsServer.ListenAndServe())
```

Comment:
I have never seen this done this way, though I guess it probably works because ListenAndServer never really returns?

In my opinion
```
if err := metricsServer.ListenAndServe(); err != nil {
  log.Fatal("Failed to start metrics server:", err)
}
``` 
is more clear.

Reply:
updated